From ce24c4e96466f82d9589f217298332b0abb2de02 Mon Sep 17 00:00:00 2001
From: Zhengyu Shen <zhengyu.shen_1@nxp.com>
Date: Mon, 18 Sep 2017 14:30:35 -0500
Subject: [PATCH 2557/5242] MLK-16359-1: Added i.MX8 CAST IP JPEG
 Encoder/Decoder Linux to DTS

commit  930d84608cd8fc0f8df8f0cb2213e63c97ca2af5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Adds device tree files for JPEG decoder and encoder to device tree.

Signed-off-by: Zhengyu Shen <zhengyu.shen_1@nxp.com>
Reviewed-by: Sandor Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |   38 +++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 78b9d34..9bc6d38 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -952,6 +952,18 @@
 				#power-domain-cells = <0>;
 				power-domains =<&pd_isi_ch0>;
 			};
+
+			pd_jpgdec: PD_IMAGING_JPEG_DEC {
+				reg = <SC_R_MJPEG_DEC_S0>;
+				#power-domain-cells = <0>;
+				power-domains =<&pd_isi_ch0>;
+			};
+
+			pd_jpgenc: PD_IMAGING_JPEG_ENC {
+				reg = <SC_R_MJPEG_ENC_S0>;
+				#power-domain-cells = <0>;
+				power-domains =<&pd_isi_ch0>;
+			};
 		};
 	};
 
@@ -1476,6 +1488,32 @@
 			power-domains = <&pd_csi1>;
 			status = "disabled";
 		};
+
+		jpegdec: jpegdec@58400000 {
+			compatible = "fsl,imx8-jpgdec";
+			reg = <0x0 0x58400000 0x0 0x00040020 >;
+			interrupts = <GIC_SPI 309 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clk IMX8QM_IMG_JPEG_DEC_IPG_CLK >,
+					<&clk IMX8QM_IMG_JPEG_DEC_CLK >;
+			clock-names = "ipg", "per";
+			assigned-clocks = <&clk IMX8QM_IMG_JPEG_DEC_IPG_CLK >,
+						<&clk IMX8QM_IMG_JPEG_DEC_CLK >;
+			assigned-clock-rates = <200000000>;
+			power-domains =<&pd_jpgdec>;
+		};
+
+		jpegenc: jpegenc@58450000 {
+			compatible = "fsl,imx8-jpgenc";
+			reg = <0x0 0x58450000 0x0 0x00240020 >;
+			interrupts = <GIC_SPI 305 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clk IMX8QM_IMG_JPEG_ENC_IPG_CLK >,
+					<&clk IMX8QM_IMG_JPEG_ENC_CLK >;
+			clock-names = "ipg", "per";
+			assigned-clocks = <&clk IMX8QM_IMG_JPEG_ENC_IPG_CLK >,
+						<&clk IMX8QM_IMG_JPEG_ENC_CLK >;
+			assigned-clock-rates = <200000000>;
+			power-domains =<&pd_jpgenc>;
+		};
 	};
 
 	i2c0: i2c@5a800000 {
-- 
1.7.9.5

